bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -0,0 +1,298 @@
+        try {
+            while (!disposed.get()) {
+                try {
+                    synchronized (pendingCommands) {
+                        pendingCommands.clear();
+                        pendingCommands.add(new MPDCommand("status"));
+                        pendingCommands.add(new MPDCommand("currentsong"));
+                    }
+
+                    establishConnection();
+                    updateThingStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, null);
+
+                    processPendingCommands();
+                } catch (IOException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } catch (MPDException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                }
+
+                isInIdle.set(false);
+                closeSocket();
+
+                if (!disposed.get()) {
+                    sleep(RECONNECTION_TIMEOUT_SEC * 1000);
+                }
+            }
+        } catch (InterruptedException e) {
+        }
+    }
+
+    /**
+     * dispose the connection
+     */
+    public void dispose() {
+        disposed.set(true);
+        Socket socket = this.socket;
+        if (socket != null) {
+            try {
+                socket.close();
+            } catch (IOException e) {
+            }
+            this.socket = null;
+        }
+    }
+
+    /**
+     * add a command to the pending commands queue
+     *
+     * @param command command to add
+     */
+    public void addCommand(MPDCommand command) {
+        insertCommand(command, -1);
+    }
+
+    private void insertCommand(MPDCommand command, int position) {
+        logger.debug("insert command '{}' at position {}", command.getCommand(), position);
+        int index = position;
+        synchronized (pendingCommands) {
+            if (index < 0) {
+                index = pendingCommands.size();
+            }
+            pendingCommands.add(index, command);
+            sendNoIdleIfInIdle();
+        }
+    }
+
+    private void updateThingStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        if (!disposed.get()) {
+            listener.updateThingStatus(status, statusDetail, description);
+        }
+    }
+
+    private void sendNoIdleIfInIdle() {
+        if (isInIdle.compareAndSet(true, false)) {
+            try {
+                sendCommand(new MPDCommand("noidle"));
+            } catch (IOException e) {
+                logger.debug("sendCommand(noidle) failed");
+            }
+        }
+    }
+
+    private void establishConnection() throws IOException, MPDException {
+        openSocket();
+
+        MPDCommand currentCommand = new MPDCommand("connect");
+        MPDResponse response = readResponse(currentCommand);
+
+        if (!response.isOk()) {
+            throw new MPDException("could not connect");
```

Comment:
It does not seem completely guaranteed to me that this is a configuration error, this could maybe just be a communication problem (daemon/host not running) however it is reported to the end-user as a configuration error with text "could not connect". Throwing it as an IOException could fix that.

```suggestion
            throw new IOException ("Failed to connect to " + this.address ":" + this.port);
```

Reply:
If the daemon or the host is not running, there is an IOException, when the socket is opened. Therefore I have left the MPDException, but I have changed the text.

Reply:
👍